### PR TITLE
test(e2e): update e2e specs for `oras attach` 

### PR DESCRIPTION
**Pull request by qweeah**
This PR make e2e specs of `oras attach` run against ZOT with proper test data.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,3 +49,10 @@ _dist/
 # Distribution storage files for local E2E testing
 test/e2e/testdata/distribution/mount/docker/
 test/e2e/testdata/distribution/mount_fallback/docker/
+
```
Comment (TerryHowe) [inline .gitignore:52]: This seems like a pain to maintain, not that I have a better idea.

Comment (qweeah) [reply .gitignore:52]: Future changes will only be required when adding or removing new repositories. E.g. If I add a new image repository for `new-test` suite, I only need to add

> !test/e2e/testdata/zot/new-test/images

Looks like I need to cover that part in the `test/e2e/README.md`

Comment (qweeah) [reply .gitignore:52]: Added test data baking guide.

File: test/e2e/README.md [hunk 2]
```diff
@@ -145,5 +146,9 @@ graph TD;
         end
     end
 ```
+
+##### Test Data for ZOT
```
Comment (Wwwsylvia) [inline test/e2e/README.md:150]: Just curious what is this for? Are the data under `testdata/zot/command/images` not "test data for ZOT"?

Comment (qweeah) [reply test/e2e/README.md:150]: Currently test data is partial, will update this section after I done migrating all the specs in command suite

**Review by Wwwsylvia — APPROVED** [review]
LGTM, but I am not a maitainer

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**